### PR TITLE
docs translations: useCallback

### LINKED ISSUE #398: `useCallback`
Page: https://id.react.dev/reference/react/useCallback

Assignee: @intanirnanda5796

### PR DESCRIPTION
**Pull request by intanirnanda5796**
translations for page: https://react.dev/reference/react/useCallback

Closes #398

### CODE REVIEW

File: src/content/reference/react/useCallback.md [hunk 1]
```diff
@@ -34,34 +34,34 @@ export default function ProductPage({ productId, referrer, theme }) {
...
+* `dependencies`: Daftar dari semua nilai yang bersifat reaktif yang direferensikan di dalam kode `fn`. Nilai-nilai reaktif termasuk props, state, dan semua variabel dan fungsi yang dideklarasikan langsung di dalam badan komponen. jika linter Anda adalah [dikonfigurasi untuk React](/learn/editor-setup#linting), linter akan memverifikasi bahwa setiap nilai reaktif telah ditentukan dengan benar sebagai dependensi. Daftar dependensi harus memiliki jumlah item yang konstan dan ditulis dalam satu baris seperti `[dep1, dep2, dep3]`. React akan membandingkan setiap dependensi dengan nilainya yang sebelumnya menggunakan algorithma perbandingan [`Object.is`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is).
```
Comment (r17x) [inline src/content/reference/react/useCallback.md:43]:
```suggestion
* `dependencies`: Daftar dari semua nilai yang bersifat reaktif yang direferensikan di dalam kode `fn`. Nilai-nilai reaktif termasuk *props*, *state*, dan semua variabel dan fungsi yang dideklarasikan langsung di dalam badan komponen. Jika *linter* Anda adalah [dikonfigurasi untuk React](/learn/editor-setup#linting), linter akan memverifikasi bahwa setiap nilai reaktif telah ditentukan dengan benar sebagai dependensi. Daftar dependensi harus memiliki jumlah *item* yang konstan dan ditulis dalam sebaris seperti `[dep1, dep2, dep3]`. React akan membandingkan setiap dependensi dengan nilainya yang sebelumnya menggunakan algoritma perbandingan [`Object.is`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is).
```

File: src/content/reference/react/useCallback.md [hunk 2]
```diff
@@ -34,34 +34,34 @@ export default function ProductPage({ productId, referrer, theme }) {
...
 
-#### Returns {/*returns*/}
+#### Pengembalian {/*returns*/}
 
-On the initial render, `useCallback` returns the `fn` function you have passed.
+Pada render awal, `useCallback` mengembalikan fungsi `fn` yang telah Anda lewati.
```
Comment (r17x) [inline src/content/reference/react/useCallback.md:47]:
```suggestion
Pada *render* awal, `useCallback` mengembalikan fungsi `fn` yang telah Anda lewati.
```

File: src/content/reference/react/useCallback.md [hunk 3]
```diff
@@ -34,34 +34,34 @@ export default function ProductPage({ productId, referrer, theme }) {
...
 
-During subsequent renders, it will either return an already stored `fn`  function from the last render (if the dependencies haven't changed), or return the `fn` function you have passed during this render.
+Selama render berikutnya, useCallback akan mengembalikan fungsi `fn` yang sudah tersimpan dari render terakhir (jika dependensi tidak berubah), atau mengembalikan fungsi `fn` yang telah Anda lewati selama render ini.
```
Comment (r17x) [inline src/content/reference/react/useCallback.md:49]:
```suggestion
Selama *render* berikutnya, useCallback akan mengembalikan fungsi `fn` yang sudah tersimpan dari *render* terakhir (jika dependensi tidak berubah), atau mengembalikan fungsi `fn` yang telah Anda lewati selama *render* ini.
```

File: src/content/reference/react/useCallback.md [hunk 4]
```diff
@@ -34,34 +34,34 @@ export default function ProductPage({ productId, referrer, theme }) {
...
+* `useCallback` adalah sebuah Hook, jadi Anda hanya dapat memanggil useCallback **di tingkat atas komponen Anda** atau Hooks yang Anda buat sendiri. Anda tidak dapat memanggil useCallback di dalam looping atau kondisi. Jika Anda membutuhkannya, ekstrak komponen baru dan pindahkan state ke dalamnya.
```
Comment (r17x) [inline src/content/reference/react/useCallback.md:53]:
```suggestion
* `useCallback` adalah sebuah Hook, jadi Anda hanya dapat memanggil useCallback **di tingkat atas komponen Anda** atau Hooks yang Anda buat sendiri. Anda tidak dapat memanggil useCallback di dalam *looping* atau kondisi. Jika Anda membutuhkannya, ekstrak komponen baru dan pindahkan *state* ke dalamnya.
```

File: src/content/reference/react/useCallback.md [hunk 5]
```diff
@@ -34,34 +34,34 @@ export default function ProductPage({ productId, referrer, theme }) {
...
+* React **tidak akan membuang fungsi yang sudah di-cache kecuali ada alasan khusus untuk melakukannya.** Sebagai contoh, pada saat pengembangan, React akan membuang cache ketika Anda mengedit file komponen Anda. Baik pada saat pengembangan maupun produksi, React akan membuang cache jika komponen Anda menunda (suspend) selama mounting awal. Di masa depan, React mungkin akan menambahkan fitur-fitur baru yang memanfaatkan pembuangan cache--Sebagai contoh, jika React menambahkan dukungan bawaan untuk virtualized lists di masa depan, maka masuk akal untuk membuang cache untuk item yang keluar dari virtualized table viewport tersebut. Hal ini seharusnya sesuai dengan ekspektasi Anda jika And mengandalkan `useCallback` sebagai optimasi kinerja. jika tidak, sebuah [state variable](/reference/react/useState#im-trying-to-set-state-to-a-function-but-it-gets-called-instead) atau sebuah [ref](/reference/react/useRef#avoiding-recreating-the-ref-contents) mungkin lebih sesuai.
```
Comment (r17x) [inline src/content/reference/react/useCallback.md:54]:
```suggestion
* React **tidak akan membuang fungsi yang sudah di-*cache* kecuali ada alasan khusus untuk melakukannya.** Sebagai contoh, pada saat pengembangan, React akan membuang *cache* ketika Anda mengedit *file* komponen Anda. Baik pada saat pengembangan maupun produksi, React akan membuang cache jika komponen Anda menunda (*suspend*) selama *mounting* awal. Di masa depan, React mungkin akan menambahkan fitur-fitur baru yang memanfaatkan pembuangan *cache*--Sebagai contoh, jika React menambahkan dukungan bawaan untuk *virtualized lists* di masa depan, maka masuk akal untuk membuang *cache* untuk *item* yang keluar dari *virtualized table viewport* tersebut. Hal ini seharusnya sesuai dengan ekspektasi Anda, Jika Anda mengandalkan `useCallback` sebagai optimasi kinerja. Jika tidak, sebuah [state variable](/reference/react/useState#im-trying-to-set-state-to-a-function-but-it-gets-called-instead) atau sebuah [ref](/reference/react/useRef#avoiding-recreating-the-ref-contents) mungkin lebih sesuai.
```

File: src/content/reference/react/useCallback.md [hunk 6]
```diff
@@ -34,34 +34,34 @@ export default function ProductPage({ productId, referrer, theme }) {
...
 ---
 
-## Usage {/*usage*/}
+## Penggunaan {/*usage*/}
 
-### Skipping re-rendering of components {/*skipping-re-rendering-of-components*/}
+### Melewati proses render ulang sebuah Komponen {/*skipping-re-rendering-of-components*/}
 
-When you optimize rendering performance, you will sometimes need to cache the functions that you pass to child components. Let's first look at the syntax for how to do this, and then see in which cases it's useful.
+ketika Anda mengoptimalkan kinerja rendering, Anda terkadang perlu meng-cache fungsi yang Anda berikan ke komponen turunan. Dan kemudian, Mari kita lihat terlebih dahulu sintaksisnya, lalu kita akan melihat dalam kasus mana ini berguna.
```
Comment (r17x) [inline src/content/reference/react/useCallback.md:62]:
```suggestion
ketika Anda mengoptimalkan kinerja *rendering*, Anda terkadang perlu meng-*cache* fungsi yang Anda berikan ke komponen turunan. Dan kemudian, Mari kita lihat terlebih dahulu sintaksisnya, lalu kita akan melihat dalam kasus mana ini berguna.
```

File: src/content/reference/react/useCallback.md [hunk 7]
```diff
@@ -34,34 +34,34 @@ export default function ProductPage({ productId, referrer, theme }) {
...
 
-To cache a function between re-renders of your component, wrap its definition into the `useCallback` Hook:
+untuk meng-cache sebuah fungsi antara render ulang dari komponen Anda, bungkus definisi fungsi Anda ke dalam Hook `useCallback`:
```
Comment (r17x) [inline src/content/reference/react/useCallback.md:64]:
```suggestion
untuk meng-*cache* sebuah fungsi antara *render* ulang dari komponen Anda, bungkus definisi fungsi Anda ke dalam Hook `useCallback`:
```

File: src/content/reference/react/useCallback.md [hunk 8]
```diff
@@ -76,20 +76,20 @@ function ProductPage({ productId, referrer, theme }) {
   // ...
 ```
 
-You need to pass two things to `useCallback`:
+Anda harus oper dua hal ke dalam `useCallback`:
 
-1. A function definition that you want to cache between re-renders.
-2. A <CodeStep step={2}>list of dependencies</CodeStep> including every value within your component that's used inside your function.
+1. Sebuah definisi fungsi yang ingin Anda cache diantara render ulang.
```
Comment (r17x) [inline src/content/reference/react/useCallback.md:81]:
```suggestion
1. Sebuah definisi fungsi yang ingin Anda *cache* diantara *render* ulang.
```

File: src/content/reference/react/useCallback.md [hunk 9]
```diff
@@ -76,20 +76,20 @@ function ProductPage({ productId, referrer, theme }) {
...
+2. Sebuah <CodeStep step={2}>list dependensi</CodeStep> termasuk setiap nilai yang ada di dalam komponen Anda yang digunakan dalam fungsi Anda.
 
-On the initial render, the <CodeStep step={3}>returned function</CodeStep> you'll get from `useCallback` will be the function you passed.
+Pada saat render awal, <CodeStep step={3}>fungsi yang dikembalikan</CodeStep> yang akan anda dapatkan dari `useCallback` akan menjadi fungsi yang anda operkan.
```
Comment (r17x) [inline src/content/reference/react/useCallback.md:84]:
```suggestion
Pada saat *render* awal, <CodeStep step={3}>fungsi yang dikembalikan</CodeStep> yang akan anda dapatkan dari `useCallback` akan menjadi fungsi yang anda operkan.
```

File: src/content/reference/react/useCallback.md [hunk 10]
```diff
@@ -76,20 +76,20 @@ function ProductPage({ productId, referrer, theme }) {
...
 
-On the following renders, React will compare the <CodeStep step={2}>dependencies</CodeStep> with the dependencies you passed during the previous render. If none of the dependencies have changed (compared with [`Object.is`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is)), `useCallback` will return the same function as before. Otherwise, `useCallback` will return the function you passed on *this* render.
+Pada render berikutnya, React akan membandingkan <CodeStep step={2}>dependensi</CodeStep> dengan dependensi yang Anda operkan selama render sebelumnya. jika tidak ada dependensi yang berubah (bandingkan dengan [`Object.is`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is)), `useCallback` akan mengembalikan fungsi yang sama seperti sebelumnya. jika tidak, `useCallback` akan mengembalikan fungsi yang telah Anda operkan di render *ini*.
```
Comment (r17x) [inline src/content/reference/react/useCallback.md:86]:
```suggestion
Pada *render* berikutnya, React akan membandingkan <CodeStep step={2}>dependensi</CodeStep> dengan dependensi yang Anda operkan selama *render* sebelumnya. Jika tidak ada dependensi yang berubah (bandingkan dengan [`Object.is`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is)), `useCallback` akan mengembalikan fungsi yang sama seperti sebelumnya. Jika tidak, `useCallback` akan mengembalikan fungsi yang telah Anda operkan di *render* ini.
```

File: src/content/reference/react/useCallback.md [hunk 11]
```diff
@@ -76,20 +76,20 @@ function ProductPage({ productId, referrer, theme }) {
...
 
-In other words, `useCallback` caches a function between re-renders until its dependencies change.
+Dengan kata lain, `useCallback` meng-cache sebuah fungsi diantara render ulang until sampai dependesi itu berubah.
```
Comment (r17x) [inline src/content/reference/react/useCallback.md:88]:
```suggestion
Dengan kata lain, `useCallback` meng-*cache* sebuah fungsi diantara *render* ulang sampai dependesi itu berubah.
```

File: src/content/reference/react/useCallback.md [hunk 12]
```diff
@@ -101,9 +101,9 @@ function ProductPage({ productId, referrer, theme }) {
   );
 ```
 
-You've noticed that toggling the `theme` prop freezes the app for a moment, but if you remove `<ShippingForm />` from your JSX, it feels fast. This tells you that it's worth trying to optimize the `ShippingForm` component.
+Anda akan menyadari bahwa mengganti nilai prop `theme` membuat aplikasi terasa berhenti sejenak, tapi jika Anda hapus `<ShippingForm />` dari JSX Anda, itu akan terasa lebih cepat. ini memberitahu Anda bahwa itu layak dicoba untuk optimasi komponen `ShippingForm`.
```
Comment (r17x) [inline src/content/reference/react/useCallback.md:104]:
```suggestion
Anda akan menyadari bahwa mengganti nilai *prop* `theme` membuat aplikasi terasa berhenti sejenak, tapi jika Anda hapus `<ShippingForm />` dari JSX Anda, itu akan terasa lebih cepat. ini memberitahu Anda bahwa itu layak dicoba untuk optimasi komponen `ShippingForm`.
```

File: src/content/reference/react/useCallback.md [hunk 13]
```diff
@@ -101,9 +101,9 @@ function ProductPage({ productId, referrer, theme }) {
...
 
-**By default, when a component re-renders, React re-renders all of its children recursively.** This is why, when `ProductPage` re-renders with a different `theme`, the `ShippingForm` component *also* re-renders. This is fine for components that don't require much calculation to re-render. But if you verified a re-render is slow, you can tell `ShippingForm` to skip re-rendering when its props are the same as on last render by wrapping it in [`memo`:](/reference/react/memo)
+**Secara default, ketika sebuah komponen melakukan render ulang, React akan melakukan render ulang pada seluruh komponen turunannya secara rekursif** inilah kenapa, ketika `ProductPage` render ulang dengan `theme` berbeda, komponen `ShippingForm` *juga* render ulang. Ini bagus untuk komponen yang tidak memerlukan banyak perhitungan untuk render ulang. tapi jika anda telah memverifikasi bahwa render ulang itu lambat, Anda dapat memberitahu `ShippingForm` untuk melewati render ulang ketika props itu sama seperti render sebelumnya dengan cara bungkus itu di dalam [`memo`:](/reference/react/memo)
```
Comment (r17x) [inline src/content/reference/react/useCallback.md:106]:
```suggestion
**Secara *default*, ketika sebuah komponen melakukan *render* ulang, React akan melakukan *render* ulang pada seluruh komponen turunannya secara rekursif** inilah kenapa, ketika `ProductPage` *render* ulang dengan `theme` berbeda, komponen `ShippingForm` juga *render* ulang. Ini bagus untuk komponen yang tidak memerlukan banyak perhitungan untuk *render* ulang. tapi jika anda telah memverifikasi bahwa *render* ulang itu lambat, Anda dapat memberitahu `ShippingForm` untuk melewati *render* ulang ketika *props* itu sama seperti *render* sebelumnya dengan cara bungkus itu di dalam [`memo`:](/reference/react/memo)
```

File: src/content/reference/react/useCallback.md [hunk 14]
```diff
@@ -113,11 +113,11 @@ const ShippingForm = memo(function ShippingForm({ onSubmit }) {
 });
 ```
 
-**With this change, `ShippingForm` will skip re-rendering if all of its props are the *same* as on the last render.** This is when caching a function becomes important! Let's say you defined `handleSubmit` without `useCallback`:
+**Dengan perubahan ini, `ShippingForm` akan melewati rendering ulang jika semua props itu *sama* seperti render sebelumnya.** inilah kenapa meng-cache sebuah fungsi menjadi penting! misalkan Anda mendefinisikan `handleSubmit` tanpa `useCallback`:
```
Comment (r17x) [inline src/content/reference/react/useCallback.md:116]:
```suggestion
**Dengan perubahan ini, `ShippingForm` akan melewati *rendering* ulang jika semua *props* itu *sama* seperti *render* sebelumnya.** inilah kenapa meng-*cache* sebuah fungsi menjadi penting! misalkan Anda mendefinisikan `handleSubmit` tanpa `useCallback`:
```

File: src/content/reference/react/useCallback.md [hunk 15]
```diff
@@ -127,59 +127,59 @@ function ProductPage({ productId, referrer, theme }) {
...
-      {/* ... so ShippingForm's props will never be the same, and it will re-render every time */}
+      {/* ... jadi prop ShippingForm tidak akan pernah sama, dan komponen tersebut akan selalu melakukan re-render setiap kali terjadi perubahan tema */}
       <ShippingForm onSubmit={handleSubmit} />
     </div>
   );
 }
 ```
 
-**In JavaScript, a `function () {}` or `() => {}` always creates a _different_ function,** similar to how the `{}` object literal always creates a new object. Normally, this wouldn't be a problem, but it means that `ShippingForm` props will never be the same, and your [`memo`](/reference/react/memo) optimization won't work. This is where `useCallback` comes in handy:
+**Di Javascript, `function () {}` atau `() => {}` akan selalu membuat sebuah fungsi _berbeda_,** serupa dengan bagaimana literal objek `{}` selalu membuat objek baru. Normalnya, ini tidak akan menjadi sebuah masalah, tetapi ini berarti bahwa props `ShippingForm` tidak akan pernah sama, dan optimasi [`memo`](/reference/react/memo) Anda tidak akan bekerja. disinilah `useCallback` berguna:
```
Comment (r17x) [inline src/content/reference/react/useCallback.md:137]:
```suggestion
**Di Javascript, `function () {}` atau `() => {}` akan selalu membuat sebuah fungsi _berbeda_,** serupa dengan bagaimana literal objek `{}` selalu membuat objek baru. Normalnya, ini tidak akan menjadi sebuah masalah, tetapi ini berarti bahwa *props* `ShippingForm` tidak akan pernah sama, dan optimasi [`memo`](/reference/react/memo) Anda tidak akan bekerja. disinilah `useCallback` berguna:
```

File: src/content/reference/react/useCallback.md [hunk 16]
```diff
@@ -127,59 +127,59 @@ function ProductPage({ productId, referrer, theme }) {
...
       <ShippingForm onSubmit={handleSubmit} />
     </div>
   );
 }
 ```
 
-**By wrapping `handleSubmit` in `useCallback`, you ensure that it's the *same* function between the re-renders** (until dependencies change). You don't *have to* wrap a function in `useCallback` unless you do it for some specific reason. In this example, the reason is that you pass it to a component wrapped in [`memo`,](/reference/react/memo) and this lets it skip re-rendering. There are other reasons you might need `useCallback` which are described further on this page.
+**Dengan membungkus `handleSubmit` dalam `useCallback`, Anda memastikan bahwa itu adalah fungsi yang *sama* antara setiap render ulang** (sampai dependensi berubah). Anda tidak *harus* bungkus sebuah fungsi dalam `useCallback`  kecuali jika Anda melakukannya untuk beberapa alasan tertentu. Dalam contoh ini, alasannya adalah Anda oper itu ke komponen yang dibungkus dalam [`memo`,](/reference/react/memo) dan ini memungkinkannya untuk lewati render ulang. ada alasan lain mengapa Anda mungkin butuh `useCallback` yang dijelaskan lebih lanjut di halaman ini.
```
Comment (r17x) [inline src/content/reference/react/useCallback.md:158]:
```suggestion
**Dengan membungkus `handleSubmit` dalam `useCallback`, Anda memastikan bahwa itu adalah fungsi yang *sama* antara setiap *render* ulang** (sampai dependensi berubah). Anda tidak *harus* bungkus sebuah fungsi dalam `useCallback`  kecuali jika Anda melakukannya untuk beberapa alasan tertentu. Dalam contoh ini, alasannya adalah Anda oper itu ke komponen yang dibungkus dalam [`memo`,](/reference/react/memo) dan ini memungkinkannya untuk lewati *render* ulang. ada alasan lain mengapa Anda mungkin butuh `useCallback` yang dijelaskan lebih lanjut di halaman ini.
```

File: src/content/reference/react/useCallback.md [hunk 17]
```diff
@@ -127,59 +127,59 @@ function ProductPage({ productId, referrer, theme }) {
...
 
 </Note>
 
 <DeepDive>
 
-#### How is useCallback related to useMemo? {/*how-is-usecallback-related-to-usememo*/}
+#### Bagaimana useCallback terkait dengan useMemo? {/*how-is-usecallback-related-to-usememo*/}
 
-You will often see [`useMemo`](/reference/react/useMemo) alongside `useCallback`. They are both useful when you're trying to optimize a child component. They let you [memoize](https://en.wikipedia.org/wiki/Memoization) (or, in other words, cache) something you're passing down:
+Anda sering melihat [`useMemo`](/reference/react/useMemo) berdampingan `useCallback`. Mereka berdua berguna ketika Anda mencoba mengoptimalkan komponen turunan. mereka memungkinkan Anda [memoize](https://en.wikipedia.org/wiki/Memoization) (atau, dengan kata lain, cache) sesuatu yang Anda telah oper ke bawah:
```
Comment (r17x) [inline src/content/reference/react/useCallback.md:170]:
```suggestion
Anda sering melihat [`useMemo`](/reference/react/useMemo) berdampingan `useCallback`. Mereka berdua berguna ketika Anda mencoba mengoptimalkan komponen turunan. mereka memungkinkan Anda [memoize](https://en.wikipedia.org/wiki/Memoization) (atau, dengan kata lain, *cache*) sesuatu yang Anda telah oper ke bawah:
```

File: src/content/reference/react/useCallback.md [hunk 18]
```diff
@@ -194,60 +194,61 @@ function ProductPage({ productId, referrer }) {
 }
 ```
 
-The difference is in *what* they're letting you cache:
+Perbedaannya terletak pada *apa* yang Anda simpan dalam cache:
```
Comment (r17x) [inline src/content/reference/react/useCallback.md:197]:
```suggestion
Perbedaannya terletak pada *apa* yang Anda simpan dalam *cache*:
```

File: src/content/reference/react/useCallback.md [hunk 19]
```diff
@@ -194,60 +194,61 @@ function ProductPage({ productId, referrer }) {
...
-* **`useCallback` caches *the function itself.*** Unlike `useMemo`, it does not call the function you provide. Instead, it caches the function you provided so that `handleSubmit` *itself* doesn't change unless `productId` or `referrer` has changed. This lets you pass the `handleSubmit` function down without unnecessarily re-rendering `ShippingForm`. Your code won't run until the user submits the form.
+* **[`useMemo`](/reference/react/useMemo) meng-cache *hasil* dari panggilan fungsi Anda.** Dalam contoh ini, itu meng-cache hasil pemanggilan fungsi `computeRequirements(product)` sehingga hasilnya tidak akan berubah kecuali `product` berubah. ini memungkinkan Anda untuk oper objek `requirements` tanpa perlu render ulang `ShippingForm`. Jika diperlukan, React akan memanggil fungsi yang telah Anda oper selama rendering untuk menghitung hasilnya.
```
Comment (r17x) [inline src/content/reference/react/useCallback.md:199]:
```suggestion
* **[`useMemo`](/reference/react/useMemo) meng-*cache* *hasil* dari panggilan fungsi Anda.** Dalam contoh ini, itu meng-*cache*  hasil pemanggilan fungsi `computeRequirements(product)` sehingga hasilnya tidak akan berubah kecuali `product` berubah. ini memungkinkan Anda untuk oper objek `requirements` tanpa perlu *render* ulang `ShippingForm`. Jika diperlukan, React akan memanggil fungsi yang telah Anda oper selama *rendering* untuk menghitung hasilnya.
```

File: src/content/reference/react/useCallback.md [hunk 20]
```diff
@@ -194,60 +194,61 @@ function ProductPage({ productId, referrer }) {
...
+* **`useCallback` meng-cache *fungsi itu sendiri.*** Berbeda dengan `useMemo`, itu tidak akan memanggil fungsi yang Anda berikan. Sebaliknya, itu akan meng-cache fungsi yang telah Anda berikan sehingga `handleSubmit` *sendiri* tidak akan berubah kecuali `productId` atau `referrer` telah berubah. Ini memungkinan Anda untuk oper fungsi `handleSubmit` ke bawah tanpa perlu render ulang `ShippingForm`. Kode Anda tidak akan dijalankan sampai pengguna submits the form.
```
Comment (r17x) [inline src/content/reference/react/useCallback.md:200]:
```suggestion
* **`useCallback` meng-cache *fungsi itu sendiri.*** Berbeda dengan `useMemo`, itu tidak akan memanggil fungsi yang Anda berikan. Sebaliknya, itu akan meng-*cache* fungsi yang telah Anda berikan sehingga `handleSubmit` *sendiri* tidak akan berubah kecuali `productId` atau `referrer` telah berubah. Ini memungkinan Anda untuk oper fungsi `handleSubmit` ke bawah tanpa perlu *render* ulang `ShippingForm`. Kode Anda tidak akan dijalankan sampai pengguna mengirimkan formulir.
```

**Review by r17x — CHANGES_REQUESTED** [review]

File: src/content/reference/react/useCallback.md (hunk 13 above)
Comment (intanirnanda5796) [reply src/content/reference/react/useCallback.md:106]: i just wanna ask something. in line 106. i translate from *also* to *juga* but from your comment become "juga". but in line 116, i translate from *same* to *sama* not become "sama" but still *sama*. so, i just wanna make sure, which is true sentences? in line 106 or 116?

**Review by r17x — APPROVED** [review]

File: src/content/reference/react/useCallback.md (hunk 13 above)
Comment (intanirnanda5796) [reply src/content/reference/react/useCallback.md]: sorry mas @r17x maybe you passed what i asked in this comment.

**Comment by mhaidarhanif** [comment]
@intanirnanda5796 Please resolve all review comments from @r17x to reconfirm.

**Comment by intanirnanda5796** [comment]
> @intanirnanda5796 Please resolve all review comments from @r17x to reconfirm.

image

i have been resolved all review comments from @r17x . but in the screenshot i have been attach, i want ask is it need to change or not?

**Comment by mhaidarhanif** [comment]
@intanirnanda5796 I meant clicking the "Resolve conversation" buttons to reconfirm. Therefore leaving only 1 conversation that still needs to be discussed or fixed. Or let's just wait for @r17x to resolve and confirm that last one conversation.

image

**Comment by intanirnanda5796** [comment]
> @intanirnanda5796 I meant clicking the "Resolve conversation" buttons to reconfirm. Therefore leaving only 1 conversation that still needs to be discussed or fixed. Or let's just wait for @r17x to resolve and confirm that remaining conversation.
> 
> image

ah oke. sorry for miss understanding.

**Comment by mhaidarhanif** [comment]
@intanirnanda5796 Alright then it will leave the last unresolved conversation: https://github.com/reactjs/id.react.dev/pull/365#discussion_r1181128855 that need to be confirmed again by @r17x

File: src/content/reference/react/useCallback.md (hunk 13 above)
Comment (r17x) [reply src/content/reference/react/useCallback.md]: I think we can remove `"` when translated yet to Bahasa, What do you think? @intanirnanda5796

Comment (intanirnanda5796) [reply src/content/reference/react/useCallback.md]: hmmm ok will do it @r17x . thanks mas.

**Comment by intanirnanda5796** [comment]
i have been push remove italic when it translate to Bahasa. if there any miss in my translate, feel free to comment mas @mhaidarhanif @r17x

**Pull request merged**